Jenkinsfile

```diff
@@ -0,0 +1,109 @@
+
+        stage('New project in OpenShift') {
+            openshift.newProject(openshiftProjectName)
+        }
+
+        openshift.withProject(openshiftProjectName) {
+            stage("Build ${operatorName} & ${operatorName}-test binaries") {
+                sh """
+                export GOOS=linux GOARCH=amd64 CGO_ENABLED=0
+                go build -o ${operatorName} cmd/manager/main.go
+                go test -c -o ${operatorName}-test ./test/e2e/...
+                """
+            }
+
+            stage("Create test file ${testFileName}") {
+                writeFile file: testFileName, text: testFileContent
+                sh "chmod +x ${testFileName}"
+            }
+
+            stage("Create a Dockerfile for ${operatorName}-test") {
+                writeFile file: "Dockerfile", text: "${dockerfileContent}"
+            }
+
+            stage("Modify operator image name in operator deployment file") {
+                sh "yq w -i deploy/operator.yaml spec.template.spec.containers[0].image ${operatorDockerImageName}"
+            }
+
+            stage("Create necessary resources") {
+                sh """
+                kubectl apply -f deploy/crds/crd.yaml -n ${openshiftProjectName} || true
+                kubectl create -f deploy/service_account.yaml -n ${openshiftProjectName}
+                kubectl create -f deploy/role.yaml -n ${openshiftProjectName}
+                kubectl create -f deploy/role_binding.yaml -n ${openshiftProjectName}
+                """
+            }
+
+            stage("Start OpenShift Build of operator image") {
+                def nb = openshift.newBuild("--name=${operatorName}-test", "--binary")
+                openshift.startBuild("${operatorName}-test", "--from-dir=.")
+                def buildSelector = nb.narrow("bc").related("builds")
+
+                try {
+                    timeout(1) {
+                        buildSelector.untilEach(1) {
+                            def buildPhase = it.object().status.phase
+                            println("Build phase:" + buildPhase)
+                            return (it.object().status.phase == "Complete")
+                        }
+                    }
+                } catch (Exception e) {
+                    buildSelector.logs()
+                    openshift.delete("project", openshiftProjectName)
+                    error "Build timed out"
+                }
+            }
+
+            stage('Test operator') {
+                try {
+                    sh "operator-sdk test cluster ${operatorDockerImageName} --namespace ${openshiftProjectName} --service-account ${operatorName}"
+                } catch (Exception e) {
+                    openshift.delete("project", openshiftProjectName)
+                    error "Test of ${operatorName} has failed."
+                }
+            }
+        }
+
+        stage('Delete OpenShift project') {
+            openshift.delete("project", openshiftProjectName)
```

Comment:
If we use Jenkins Declarative Pipeline we can use the post always section [1] to delete the project in case of error and success without having to repeat this line in every try catch

[1] https://jenkins.io/doc/book/pipeline/syntax/#post